tests/functionnal/RuleTicket.php

```diff
@@ -787,7 +787,139 @@ public function testTaskTemplateAssignFromRule() {
       $this->string($task_data['content'])->isEqualTo('test content');
    }
 
-   public function testGroupRequesterAssignFromUserGroupsAndRegex() {
+   public function testGroupRequesterAssignFromUserGroupsAndRegexOnUpdateTicketContent() {
+      $this->login();
+
+      // Create rule
+      $ruleticket = new \RuleTicket();
+      $rulecrit   = new \RuleCriteria();
+      $ruleaction = new \RuleAction();
+
+      $ruletid = $ruleticket->add($ruletinput = [
+         'name'         => 'test regex group requester criterion',
+         'match'        => 'AND',
+         'is_active'    => 1,
+         'sub_type'     => 'RuleTicket',
+         'condition'    => \RuleTicket::ONUPDATE,
+         'is_recursive' => 0,
+      ]);
+      $this->checkInput($ruleticket, $ruletid, $ruletinput);
+
+      //create criteria to check if group requester match regex
+      $crit_id = $rulecrit->add($crit_input = [
+         'rules_id'  => $ruletid,
+         'criteria'  => '_groups_id_of_requester',
+         'condition' => \Rule::REGEX_MATCH,
+         'pattern'   => Toolbox::addslashes_deep('/(.+\([^()]*\))/'),   //retrieve groupe with '(' and ')'
+      ]);
+      //change value because of addslashes
+      $crit_input['pattern'] = '/(.+\([^()]*\))/';
+      $this->checkInput($rulecrit, $crit_id, $crit_input);
+
+      //create action to put  group matching on criteria
+      $action_id = $ruleaction->add($action_input = [
+         'rules_id'    => $ruletid,
+         'action_type' => 'regex_result',
+         'field'       => '_groups_id_requester',
+         'value'       => '#0',
+      ]);
+      $this->checkInput($ruleaction, $action_id, $action_input);
+
+      //Load user post_only
+      $user = new \User();
+      $user->getFromDB(getItemByTypeName('User', 'post-only', true));
+
+      //create 2 new group
+      $group = new \Group();
+      $group_id1 = $group->add($group_input = [
+         "name"         => "group1 (5215)",
+         "is_requester" => true
+      ]);
+      $this->checkInput($group, $group_id1, $group_input);
+
+      //create new group
+      $group_id2 = $group->add($group_input = [
+         "name"         => "group2 (13)",
+         "is_requester" => true
+      ]);
+      $this->checkInput($group, $group_id2, $group_input);
+
+      // create ticket that trigger rule on creation
+      $ticket = new \Ticket();
+      $ticket->getEmpty();
+      $tickets_id = $ticket->add($ticket_input = [
+         'name'                  => 'Add group requester',
+         'id'                    => 0, //pass id = 0 because in IHM it's passed as POST
+         'content'               => 'test',
+         '_users_id_requester'   => $user->fields['id']
+      ]);
+      unset($ticket_input['_users_id_requester']);
+      $ticket_input['id'] = $tickets_id; //force id to $input
```

Comment:
Not required if previous suggestion is applied.
```suggestion
```

Or an alternative solution is:
```suggestion
      unset($ticket_input['id'];
```

Reply:
done